pkg/ssh/ssh.go

```diff
@@ -291,6 +292,33 @@ func ParsePublicKeyFromFile(keyFile string) (*rsa.PublicKey, error) {
 	return rsaKey, nil
 }
 
+func MakePrivateKeySignerFromAgent(keyComment string) (ssh.Signer, error) {
+	sockName := os.Getenv("SSH_AUTH_SOCK")
+	if len(sockName) == 0 {
+		return nil, fmt.Errorf("SSH_AUTH_SOCK not defined in the environment")
+	}
+	socket, err := net.DialTimeout("unix", sockName, time.Minute)
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to agent: %v", err)
+	}
+	signers, err := agent.NewClient(socket).Signers()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve keys from agent: %v", err)
+	}
+	keyCommentSuffix := " " + keyComment
+	for _, signer := range signers {
+		agentKey, ok := signer.PublicKey().(*agent.Key)
+		if !ok {
+			continue
+		}
+		serialized := agentKey.String()
+		if strings.HasSuffix(serialized, keyCommentSuffix) {
+			return signer, nil
+		}
+	}
+	return nil, fmt.Errorf("key with comment '%s' not found among %d keys in agent", keyComment, len(signers))
```

Comment:
You could use `%q` instead of `'%s'` to put `keyComment` in quotes, if you'd like.

Reply:
Nice, thanks!